### PR TITLE
Add option to keep caret inside block in edit mode

### LINKED ISSUE #22190: Edit mode: arrow up/down should not move to other blocks
When using a screen reader in Gutenberg edit mode, pressing up/down arrow while in a block can cause the blocks to change. E.g. let's say I have a heading, paragraph, and quote on the page. If I am busy typing a paragraph out in edit mode and then want to go back and read what I have so far, I might press CTRL+HOME to go to the beginning then down arrow to read line by line. The problem becomes if I reach the end of the content and don't realize it, I could switch to the quote block and not even realize it.

I would like to see this feature adapted that way you must exit edit mode by pressing Escape before you can change blocks. This way the user knows exactly which block they are in at all times. This would be similar to how the Voiceover screen reader works on Mac OSX. When you want to perform a specific action or explore a structured element, you interact with that element and nothing else. I am thinking this type of approach would work really well for editing content in blocks that way a user never gets confused as to where they are from an accidental press of the up/down arrow when at the top/bottom of the block.

One possible alternative would be to add a setting that allows the user to choose whether or not to trap focus in the block or not based on if the user is in edit mode or not. This would be a good solution as well since this would allow for the user to make the decision on what they feel most comfortable with.

Thoughts, opposing opinions?

### PR DESCRIPTION
**Pull request by tellthemachines**
## Description

Fixes #22190, builds on #22204.

Added "Keyboard options" section in "More tools & options > Options" modal, with checkbox labelled "Keep cursor inside active block". Checking it will stop arrow keys from navigating between blocks in edit mode. Navigation mode is not affected.

## How has this been tested?

Tested in browser with keyboard.

## Types of changes

New feature (non-breaking change which adds functionality)

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/edit-post/src/components/options-modal/index.js [hunk 1]
```diff
@@ -55,6 +56,11 @@ export function OptionsModal( { isModalActive, isViewable, closeModal } ) {
 					) }
 				/>
 			</Section>
+			<Section title={ __( 'Keyboard options' ) }>
+				<EnableKeepCaretInBlock
```
Comment (youknowriad) [inline packages/edit-post/src/components/options-modal/index.js:60]: Can we use `EnableFeature` instead to enaable persisting the setting in local storage and update the EditorProvider to pass down that as a setting to `block-editor`  instead? (similar to `mostUsedBlocks` for instance)

**Comment by enriquesanchez** [comment]
Hi @tellthemachines 👋

I just tested this PR and everything worked as expected to me.

Screen Shot 2020-06-29 at 17 52 44

I saw the new option and after turning it on, I was able to confirm that the caret stays constrained in the block. Turning it back off took me back to the default behaviour.

**Review by youknowriad — APPROVED** [review]
Code wise this is looking good.

cc @mtias @jasmussen do you have thoughts here?

**Comment by jasmussen** [comment]
This seems good. 

On the one hand, it's good that the label is short and simple. But on the other hand, this is quite a profound change to the editor — it may be exactly what you're looking for, or it may break your experience entirely. I wonder if there's a more descriptive label we can find, or whether this may even be a place to add help text. "Cursor", for example, may even be imprecise, as you can move your cursor anywhere. 

Short option:

	[ ] Contain text cursor inside active block

This is a bit more technical, actually, so not sure it's an improvement. But it feels more precise.

Longer option:

	[ ] Contain text cursor inside active block
	_This option can benefit users of assistive technologies._

Regardless of what we go with, I would say it's good to merge sooner rather than later, because it's a good option and we can alwasy polish the label if need be.

**Comment by alexstine** [comment]
Tested using NV Access in Firefox, works fine.

**Comment by tellthemachines** [comment]
@jasmussen I went with the short option for now; we can always review it later.

Thanks everyone for your feedback!

**Pull request merged**